src/DynamoCore/Models/RecordableCommands.cs

```diff
@@ -689,6 +689,110 @@ protected override void SerializeCore(XmlElement element)
+                SetProperties(outputPortIndex, inputPortIndex, x, y, defaultPosition, transformCoordinates);
+            }
+
+            #endregion
+
+            #region Public Command Properties
+
+            // Faster, direct creation
+            internal NodeModel InputNode { get; private set; }
+
+            internal NodeModel GivenNode { get; private set; }
+
+            [DataMember]
+            internal int OutputPortIndex { get; private set; }
+            
+            [DataMember]
+            internal int InputPortIndex { get; private set; }
+
+
+
+            [DataMember]
+            internal double X { get; private set; }
+
+            [DataMember]
+            internal double Y { get; private set; }
+
+            [DataMember]
+            internal bool DefaultPosition { get; private set; }
+
+            [DataMember]
+            internal bool TransformCoordinates { get; private set; }
+
+            [DataMember]
+            //Legacy properties
+            public string Name { get; private set; }
+
+            #endregion
+
+            #region Protected Overridable Methods
+
+            protected override void ExecuteCore(DynamoModel dynamoModel)
+            {
+                //dynamoModel.AddNodeToCurrentWorkspace(InputNode, centered: false, addToSelection: false);
+                dynamoModel.CurrentWorkspace.RecordCreatedModel(InputNode);
+
+                var mode = MakeConnectionCommand.Mode.Begin;
+                var cmd = new MakeConnectionCommand(InputNode.GUID, OutputPortIndex, PortType.Output, mode);
+                dynamoModel.ExecuteCommand(cmd);
+
+                mode = MakeConnectionCommand.Mode.End;
+                cmd = new MakeConnectionCommand(GivenNode.GUID, InputPortIndex, PortType.Input, mode);
+                dynamoModel.ExecuteCommand(cmd);
+            }
+
+            protected override void SerializeCore(XmlElement element)
+            {
+                base.SerializeCore(element);
+                var helper = new XmlElementHelper(element);
+                helper.SetAttribute("X", X);
+                helper.SetAttribute("Y", Y);
+                helper.SetAttribute("DefaultPosition", DefaultPosition);
+                helper.SetAttribute("TransformCoordinates", TransformCoordinates);
+
+                if (InputNode != null)
+                {
+                    var nodeElement = InputNode.Serialize(element.OwnerDocument, SaveContext.File);
+                    element.AppendChild(nodeElement);
+                }
+                helper.SetAttribute("PortIndex", OutputPortIndex);
+                helper.SetAttribute("Type", ((int)PortType.Output));
+                helper.SetAttribute("ConnectionMode", ((int)MakeConnectionCommand.Mode.Begin));
+
+                helper.SetAttribute("PortIndex", InputPortIndex);
+                helper.SetAttribute("Type", ((int)PortType.Input));
+                helper.SetAttribute("ConnectionMode", ((int)MakeConnectionCommand.Mode.End));
```

Comment:
@Benglin thanks for your comments. I don't see why we need a `DeserializeCore` method so didn't include it. This being a composite command with a `CreateNodeCommand` followed by two `MakeConnectionCommand`s, I think if it is serialized to a file, it can easily be deserialized using the `DeserializeCore` methods of the individual commands. Am I right?